# Tasks

In [3]:
# Data frames
import pandas as pd

# Machine learning Linrary that contains sample datasets
import sklearn as skl

#Plots - week6
import matplotlib.pyplot as plt

#Numerical arrays.
import numpy as np

## Task 1: Source the Data Set

Import the Iris data set from the sklearn.datasets module.
Explain, in your own words, what the load_iris() function returns.
https://gist.githubusercontent.com/curran/a08a1080b88344b0c8a7/raw/0e7a9b0a5d22642a06d3d5b9bcbad9890c8ee534/iris.csv

scikit-learn
https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_iris.html#

In [4]:
#Load the dataset
data = skl.datasets.load_iris()

## Task 1 - load_iris() - Comments
This function loads the raw csv file from Scklit-Learn library into the Jupyter notebook. We can further analyse the dataset once imported.

## Task 2: Explore the Data Structure
Print and explain the
1. shape of the data set
2. first and last 5 rows of the data
3. feature names
4. the target classes
df. describe () - this is used to explore the data


In [5]:
#Shape of Iris Dataset
data ['data'].shape

(150, 4)

## Task 2 Shape of the Dataset - Comments
Iris dataset has a shape of 150,4 , meaning the dataset contains:
- 150 rows detailing 150 samples of iris flowers
- 4 columns(feautures) that describe the characterisitics of the iris flowers

4 Columns represent:

1. sepal length (cm)
2. sepal width (cm)
3. petal length (cm)
4. petal width (cm)


Reference - "https://www.geeksforgeeks.org/iris-dataset/"
Reference - "https://gist.github.com/curran/a08a1080b88344b0c8a7"

In [6]:
#Feature Name
data['feature_names']

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

In [7]:
#look at the data
data

{'data': array([[5.1, 3.5, 1.4, 0.2],
        [4.9, 3. , 1.4, 0.2],
        [4.7, 3.2, 1.3, 0.2],
        [4.6, 3.1, 1.5, 0.2],
        [5. , 3.6, 1.4, 0.2],
        [5.4, 3.9, 1.7, 0.4],
        [4.6, 3.4, 1.4, 0.3],
        [5. , 3.4, 1.5, 0.2],
        [4.4, 2.9, 1.4, 0.2],
        [4.9, 3.1, 1.5, 0.1],
        [5.4, 3.7, 1.5, 0.2],
        [4.8, 3.4, 1.6, 0.2],
        [4.8, 3. , 1.4, 0.1],
        [4.3, 3. , 1.1, 0.1],
        [5.8, 4. , 1.2, 0.2],
        [5.7, 4.4, 1.5, 0.4],
        [5.4, 3.9, 1.3, 0.4],
        [5.1, 3.5, 1.4, 0.3],
        [5.7, 3.8, 1.7, 0.3],
        [5.1, 3.8, 1.5, 0.3],
        [5.4, 3.4, 1.7, 0.2],
        [5.1, 3.7, 1.5, 0.4],
        [4.6, 3.6, 1. , 0.2],
        [5.1, 3.3, 1.7, 0.5],
        [4.8, 3.4, 1.9, 0.2],
        [5. , 3. , 1.6, 0.2],
        [5. , 3.4, 1.6, 0.4],
        [5.2, 3.5, 1.5, 0.2],
        [5.2, 3.4, 1.4, 0.2],
        [4.7, 3.2, 1.6, 0.2],
        [4.8, 3.1, 1.6, 0.2],
        [5.4, 3.4, 1.5, 0.4],
        [5.2, 4.1, 1.5, 0.1],
  

In [9]:
#Look at the data keys
data.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])

In [10]:
#The so call features
data ['data']

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2],
       [5.4, 3.9, 1.7, 0.4],
       [4.6, 3.4, 1.4, 0.3],
       [5. , 3.4, 1.5, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [4.9, 3.1, 1.5, 0.1],
       [5.4, 3.7, 1.5, 0.2],
       [4.8, 3.4, 1.6, 0.2],
       [4.8, 3. , 1.4, 0.1],
       [4.3, 3. , 1.1, 0.1],
       [5.8, 4. , 1.2, 0.2],
       [5.7, 4.4, 1.5, 0.4],
       [5.4, 3.9, 1.3, 0.4],
       [5.1, 3.5, 1.4, 0.3],
       [5.7, 3.8, 1.7, 0.3],
       [5.1, 3.8, 1.5, 0.3],
       [5.4, 3.4, 1.7, 0.2],
       [5.1, 3.7, 1.5, 0.4],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.3, 1.7, 0.5],
       [4.8, 3.4, 1.9, 0.2],
       [5. , 3. , 1.6, 0.2],
       [5. , 3.4, 1.6, 0.4],
       [5.2, 3.5, 1.5, 0.2],
       [5.2, 3.4, 1.4, 0.2],
       [4.7, 3.2, 1.6, 0.2],
       [4.8, 3.1, 1.6, 0.2],
       [5.4, 3.4, 1.5, 0.4],
       [5.2, 4.1, 1.5, 0.1],
       [5.5, 4.2, 1.4, 0.2],
       [4.9, 3

In [12]:
data ['target']

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])

In [13]:
#The target names
data['target_names']

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

## Task 3: Summarize the Data

For each feature in the dataset, calculate and display:

- mean
- minimum
- maximum
- standard deviation
- median

In [ ]:
#Converted from sklearn to numpy
data_array = np.array(data.data)



NameError: name 'features' is not defined

In [ ]:
# Convert to numpy

#mean
data_mean = np.mean(data_array , axis = 0) # axis = 0 Calculate the mean of each column # https://stackoverflow.com/questions/40200070/what-does-axis-0-do-in-numpys-sum-function
print("Column Mean:", data_mean)

Column Mean: [5.84333333 3.05733333 3.758      1.19933333]


## Task 4: Visualize Features
#### Plot histograms for each feature using matplotlib. Add appropriate titles and axis labels.


## Task 5: Investigate Relationships

#### Choose any two features from the data set and create a scatter plot of them.
#### Color-code the three different classes of the scatter plot points.

## Task 6: Analyze Relationship

#### Use numpy.polyfit to add a regression line to the scatter plot from Task 5.

## Task 7: Analyze Class Distributions

#### Create box-plots of the petal lengths for each of the three classes.

## Task 8: Compute Correlations

#### Calculate the correlation coefficients between the features.
#### Display the results as a heatmap using matplotlib.

## Task 9: Fit a Simple Linear Regression

#### For your two features in Task 5, calculate the coefficient of determination r²
#### Re-create the plot from Task 6 and annotate it with the r² value


## Task 10: Too Many Features

##### Use seaborn to create a pairplot of the data set.
##### Explain, in your own words, what the pairplot depicts

## End